ada/ast.py

```diff
@@ -8478,13 +8479,65 @@ def xref_equation():
             # in `Convention => C` to the first visible entity named C.
             Entity.id.name_is('Convention'), LogicTrue(),
 
+            Entity.id.name_is('Stable_Properties'),
+            Entity.stable_properties_assoc_equation,
+
             # Default resolution: For the moment we didn't encode specific
             # resolution rules for every aspect, so by default at least try to
             # name resolve the expression.
             Entity.expr.then(lambda e: e.sub_equation, default_val=LogicTrue())
             | LogicTrue()
         )
 
+    @langkit_property(return_type=Equation, dynamic_vars=[env, origin])
+    def stable_properties_assoc_equation():
+        """
+        Equation for the case where this is an aspect assoc for a
+        Stable_Properties aspect.
+        """
+        # Get the list of names defined by the aspect
+        identifiers = Var(
+            Entity.expr.match(
+                # AspectAssoc is of the form: (name1, name2, ...)
+                lambda a=T.Aggregate: a.assocs.map(lambda i: i.expr),
+                # AspectAssoc is of the form: (name)
+                lambda pe=T.ParenExpr: pe.expr.singleton,
+                lambda _: No(T.Expr.entity).singleton
+            ).map(
+                lambda e:
+                # Ignore the `not` keyword (useless for nameres)
+                e.cast(T.UnOp).then(
+                    lambda uo: If(uo.op.is_a(Op.alt_not), uo.expr, e),
+                    default_val=e
+                )
+            ).map(
+                # Names defined by the assoc can only be `Identifier`s
+                lambda e: e.cast_or_raise(T.Identifier)
+            )
+        )
+
+        return identifiers.logic_all(
+            lambda i: Self.env_get(
+                env,
+                i.sym,
+                lookup=LK.recursive,
+                from_node=Self.origin_node,
+                categories=all_categories
+            ).filtermap(
+                # It can only refers to a SubpDecl
```

Comment:
`refers` -> `refer`

Reply:
Also, `SubpDecl` does not include `ExprFunction`, is that OK according to the AI?

Reply:
hm, I don't see anything that would forbid their use. If GNAT supports them, I'll fix the PR accordingly.